llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -592,8 +662,223 @@ static Error applyAddends(MCCASReader &Reader,
+      FormSize += (FormVal == dwarf::DW_FORM_addr) ? FP.AddrSize
+                                                   : FP.getRefAddrByteSize();
+      break;
+    }
+    case dwarf::DW_FORM_exprloc:
+    case dwarf::DW_FORM_block: {
+      DWARFDataExtractor DWARFExtractor(toStringRef(CUData), IsLittleEndian,
+                                        AddressSize);
+      uint64_t PrevOffset = CUOffset;
+      CUOffset += DWARFExtractor.getULEB128(&CUOffset);
+      FormSize += CUOffset - PrevOffset;
+      break;
+    }
+    case dwarf::DW_FORM_block1: {
+      DWARFDataExtractor DWARFExtractor(toStringRef(CUData), IsLittleEndian,
+                                        AddressSize);
+      uint64_t PrevOffset = CUOffset;
+      CUOffset += DWARFExtractor.getU8(&CUOffset);
+      FormSize += CUOffset - PrevOffset;
+      break;
+    }
+    case dwarf::DW_FORM_block2: {
+      DWARFDataExtractor DWARFExtractor(toStringRef(CUData), IsLittleEndian,
+                                        AddressSize);
+      uint64_t PrevOffset = CUOffset;
+      CUOffset += DWARFExtractor.getU16(&CUOffset);
+      FormSize += CUOffset - PrevOffset;
+      break;
+    }
+    case dwarf::DW_FORM_block4: {
+      DWARFDataExtractor DWARFExtractor(toStringRef(CUData), IsLittleEndian,
+                                        AddressSize);
+      uint64_t PrevOffset = CUOffset;
+      CUOffset += DWARFExtractor.getU32(&CUOffset);
+      FormSize += CUOffset - PrevOffset;
+      break;
+    }
+    case dwarf::DW_FORM_data1:
+    case dwarf::DW_FORM_ref1:
+    case dwarf::DW_FORM_flag:
+    case dwarf::DW_FORM_strx1:
+    case dwarf::DW_FORM_addrx1: {
+      FormSize += 1;
+      break;
+    }
+    case dwarf::DW_FORM_data2:
+    case dwarf::DW_FORM_ref2:
+    case dwarf::DW_FORM_strx2:
+    case dwarf::DW_FORM_addrx2: {
+      FormSize += 2;
+      break;
+    }
+    case dwarf::DW_FORM_strx3: {
+      FormSize += 3;
+      break;
+    }
+    case dwarf::DW_FORM_data4:
+    case dwarf::DW_FORM_ref4:
+    case dwarf::DW_FORM_ref_sup4:
+    case dwarf::DW_FORM_strx4:
+    case dwarf::DW_FORM_addrx4: {
+      FormSize += 4;
+      break;
+    }
+    case dwarf::DW_FORM_data8:
+    case dwarf::DW_FORM_ref8:
+    case dwarf::DW_FORM_ref_sup8: {
+      FormSize += 8;
+      break;
+    }
+    case dwarf::DW_FORM_data16: {
+      FormSize += 16;
+      break;
+    }
+    case dwarf::DW_FORM_sdata: {
+      DWARFDataExtractor DWARFExtractor(toStringRef(CUData), IsLittleEndian,
+                                        AddressSize);
+      uint64_t PrevOffset = CUOffset;
+      DWARFExtractor.getSLEB128(&CUOffset);
+      FormSize += CUOffset - PrevOffset;
+      break;
+    }
+    case dwarf::DW_FORM_udata:
+    case dwarf::DW_FORM_ref_udata:
+    case dwarf::DW_FORM_rnglistx:
+    case dwarf::DW_FORM_loclistx:
+    case dwarf::DW_FORM_GNU_addr_index:
+    case dwarf::DW_FORM_GNU_str_index:
+    case dwarf::DW_FORM_addrx:
+    case dwarf::DW_FORM_strx: {
```

Comment:
could DW_FORM_block be handled here, too? It looks similar

Reply:
Nope, they are different in a DW_FORM_block, the ULEB128 value is the number of bytes we need to read to get the actual block contents. 

So you have a ULEB128 value that you read, that tells you how many more bytes to read, and that is the actual block, but in the case of addrx, strx etc the ULEB128 value is the value of the addrx, strx etc. 

I hope that makes sense?